frontend/subscriptions/index.js

```diff
@@ -11,14 +12,22 @@ const groupedProductsSubscriptions = (subscribe) => {
   /**
    * Gets triggered on entering the product details route.
    */
-  subscribe(productIsReady$, ({ dispatch, getState }) => {
-    if (hasGroupedProducts(getState())) {
-      dispatch(hideAddToCartBar());
-      dispatch(getProductChildren());
+  subscribe(receivedVisibleProduct$, ({
+    dispatch, getState, action,
+  }) => {
+    const productId = action.productData.id;
+    if (productId === null) {
+      return;
+    }
+    if (hasGroupedProducts(getState(), { productId })) {
+      dispatch(getProductChildren(productId));
     } else {
       dispatch(showAddToCartBar());
     }
   });
+  subscribe(productChildrenReceived$, ({ dispatch }) => {
+    dispatch(hideAddToCartBar());
```

Comment:
Why wait until the child products are received to hide the add to cart bar?  (this is just a question)